sdcm/tester.py

```diff
@@ -1716,6 +1717,60 @@ def destroy_credentials(self):  # pylint: disable=no-self-use
                 credential.destroy()
             self.credentials = []
 
+    @silence()
+    def show_alive_threads(self):
+        if not threading.active_count():
+            return
+        self.log.error("There are some threads left alive:")
+        for th_num, th in enumerate(threading.enumerate()):
+            if th is threading.current_thread():
+                continue
+            source = '<no code available>'
+            module = 'Unknown'
+            if th.__class__ is threading.Thread:
+                if th.run.__func__ is not threading.Thread.run:
+                    module = th.run.__module__
+                    source = inspect.getsource(th.run)
+                elif getattr(th, '_target', None):
+                    module = th._target.__module__
+                    source = inspect.getsource(th._target)
+            else:
+                module = th.__module__
+                source = inspect.getsource(th.__class__)
+            self.log.error(f"========= Thread {th.name} from {module} =========")
+            self.log.error(f"========= SOURCE =========\n{source}")
+            self.log.error(f"========= STACK TRACE =========\n{self._get_stacktrace(th)}")
+            self.log.error(f"========= END OF Thread {th.name} from {module} =========")
+
+    def _get_stacktrace(self, thread):
+        frame = sys._current_frames().get(thread.ident, None)
+        output = []
+        for filename, lineno, name, line in traceback.extract_stack(frame):
+            output.append('File: "%s", line %d, in %s' % (filename,
+                                                          lineno, name))
+            if line:
+                output.append("  %s" % (line.strip()))
+        return '\n'.join(output)
+
+    @silence()
```

Comment:
did you test this? When exception will happen you will fail to publish TestFramewrokEvent since you run the silence decorator when the events device is stopped...

Reply:
per our discussion `publish_or_dump` supports the case when events device is stopped